### PR TITLE
Use Click instead of Flask-Script for CLI commands

### PR DESCRIPTION
**Pull request by Jaza**
- Update Flask to 0.11 (with built-in Click integration)
- Remove manage.py
- Add autoapp.py per Flask doc's new recommendations
  for using Click with an app factory
- Define cookiecutter-flask's commands in cli.py
- Register shell context and shell commands as part
  of factory-based app creation
- Update README instructions with examples for running
  commands in new Click style

### CODE REVIEW

**Comment by Jaza** [comment]
Trying to address #76 with this PR. Note: I don't recommend merging this yet, it's not ready, todo list for this includes:

- Get Flask-Script's `Clean()` and `ShowUrls()` working with Click (ideally in a separate project that branches off Flask-Script, to keep these maintained separately, although could be done here in cookiecutter-flask)
- Get `Lint()` working with Click (probably makes sense to keep this within cookiecutter-flask)

**Comment by Jaza** [comment]
OK, now the `lint`, `clean`, and `urls` commands are included in this PR, should be ready to merge. Let me know if you think anything is missing.

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/commands.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+PROJECT_ROOT = os.path.join(HERE, os.pardir)
+TEST_PATH = os.path.join(PROJECT_ROOT, 'tests')
+
+
+@click.command()
+def test():
+    """Run the tests."""
+    import pytest
+    rv = pytest.main([TEST_PATH, '--verbose'])
+    if rv is not 0:
```
Comment (sloria) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/commands.py]: Minor: This check is unnecessary.

File: {{cookiecutter.app_name}}/README.rst [hunk 2]
```diff
@@ -50,32 +47,30 @@ Shell
 
 To open the interactive shell, run ::
 
-    python manage.py shell
+    flask shell
 
-By default, you will have access to ``app``, ``db``, and the ``User`` model.
+By default, you will have access to ``app``, ``db``, ``g``, and the ``User`` model.
```
Comment (sloria) [inline {{cookiecutter.app_name}}/README.rst]: This isn't quite accurate. The default `flask shell` command only gives you access to the `app`.

**Comment by sloria** [comment]
Thanks a lot for this PR! I addressed the minor nitpicks when I merged.

**Comment by Jaza** [comment]
No problem @sloria , thanks for merging - now to Click-ify the rest of my `cookiecutter-flask` based apps...

**Pull request merged**